include/swift/AST/TensorFlow.h

```diff
@@ -92,7 +92,13 @@ namespace tf {
     bool containsTensorFlowValue(Type ty, bool checkHigherOrderFunctions);
 
   private:
-    bool structContainsTensorFlowValue(StructDecl *decl);
+    bool containsTensorFlowValueImpl(
```

Comment:
No, it is not a cache. It keeps track of the nesting structure so that we can detect recursive data structures. 

In any case, ignore the file in this PR. I have opened a separate PR for this: https://github.com/apple/swift/pull/21449

(I had to include the files here to make sure I can run the tests.)